openfasoc/generators/common/simulation/classify_sim_error.py

```diff
@@ -0,0 +1,55 @@
+import warnings
+import sys
+import os
+
+sys.path.append(os.path.join(os.path.dirname(__file__), '..'))
+from common.get_ngspice_version import check_ngspice_version
+            
+def classify_sim_error(template_file, result_file, _generator_is) -> str:
+    """Used to propogate out how close the generated simulations results are 
+    from the results in the stored template files
+
+    Args:
+    - 'template_file' (string): The stored template file
+    - 'result_file' (string): The result file generated by the simulations
+    - '_generator_is' (dict): Contains info about the generator for which the simulation check is being run
+
+    Returns:
+    - str: The kind of alert
+        - 'red' alert for very large difference in generated and template results
```

Comment:
Can this return a percentage error instead? Then, each generator could have a different margin and the logs would display a useful metric.

Reply:
Ammended so that the generator specific `compare_files` function throws more detailed errors alongwith passing the error code